fuel/src/main/kotlin/com/github/kittinunf/fuel/core/Encoding.kt

```diff
@@ -32,26 +29,22 @@ class Encoding(
                 }
                 modifiedPath += (querySign + queryParamString)
             }
-            requestType == Request.Type.UPLOAD -> {
-                val boundary = System.currentTimeMillis().toString(16)
-                headerPairs[Headers.CONTENT_TYPE] = "multipart/form-data; boundary=$boundary"
-            }
             else -> {
                 headerPairs[Headers.CONTENT_TYPE] = "application/x-www-form-urlencoded"
                 data = queryFromParameters(parameters)
             }
         }
-        Request(
-                method = method,
-                path = modifiedPath,
-                url = createUrl(modifiedPath),
-                type = requestType,
-                parameters = parameters ?: emptyList(),
-                timeoutInMillisecond = timeoutInMillisecond,
-                timeoutReadInMillisecond = timeoutReadInMillisecond,
-                headers = headerPairs
-        ).apply {
-            if (data != null) body(data)
+        RegularRequest(
+            method = method,
+            path = modifiedPath,
+            url = createUrl(modifiedPath),
+            parameters = parameters ?: emptyList(),
+            headers = headerPairs
+        ).let {
+            when (data) {
```

Comment:
would an `if` be more readable?

Reply:
Personally I like the explicit return. I think this should actually be wrapped in a function `applyBody`, then we can use an if.